### PR TITLE
chore: auto generate changelist

### PR DESCRIPTION
**Pull request by sethkfman**
## **Description**

This PR creates a yarn command that does two things:
1.  creates csv file on the change diff between commits and categorizes by change type based off of the auto-changelog format
2. updates the changelog with the diff categorized by Added, Change, Fixed, Ops, Unknown 

## **Related issues**

Fixes: 

## **Manual testing steps**

1. Pull the branch down
2. Run `yarn setup`
3. Run `yarn changeset-documentation <base branch/tag/hash> <target branch/tag/hash>`
4. Observe output of `commits.csv` at root level and modified `changelog.md` file

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by sethkfman** [comment]
@SocketSecurity ignore npm/simple-git@3.22.0 

Shell access is expected

File: package.json [hunk 1]
```diff
@@ -76,6 +76,7 @@
     "test:merge-coverage": "nyc report --temp-dir ./tests/coverage --report-dir ./tests/merged-coverage/ --reporter json --reporter text --reporter lcovonly",
     "test:validate-coverage": "nyc check-coverage --nycrc-path ./coverage-thresholds.json -t ./tests/merged-coverage/",
     "update-changelog": "./scripts/auto-changelog.sh",
+    "changeset-changelog": "wrap () { node ./scripts/generate-rc-commits.js \"$@\" | cat  && ./scripts/changelog-csv.sh; }; wrap ",
```
Comment (legobeat) [inline package.json:79]:
```suggestion
    "changeset-changelog": "wrap () { node ./scripts/generate-rc-commits.js \"$@\"  && ./scripts/changelog-csv.sh; }; wrap ",
```
http://www.novosial.org/shell/useless-cat/

File: scripts/changelog-csv.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    # Replace the original CHANGELOG with the updated one
+    mv "$temp_changelog" "$changelog"
+}
+
+# Insert new entries into CHANGELOG.md
+insert_new_entries
+
+# Cleanup
+rm "$NEW_ENTRIES-"* "$CHANGELOG_BACKUP"
```
Comment (legobeat) [inline scripts/changelog-csv.sh:73]: Could put the cleanup in a trap to make it happen even on unclean exit of the script

http://redsymbol.net/articles/bash-exit-traps/

File: scripts/changelog-csv.sh [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+echo 'CHANGELOG updated'
```
Comment (legobeat) [inline scripts/changelog-csv.sh:75]: nit: missing newline at EOF

File: scripts/changelog-csv.sh [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+set -e
+set -u
+set -o pipefail
+
+readonly CSV_FILE='commits.csv'
+# Temporary file for new entries
+NEW_ENTRIES=$(mktemp)
+# Backup file for existing CHANGELOG
+CHANGELOG_BACKUP="CHANGELOG.md.bak"
```
Comment (legobeat) [inline scripts/changelog-csv.sh:11]:
```suggestion
CHANGELOG_BACKUP="CHANGELOG.md.bak-$(date +%s)"
```
this should make it unique that you don't wipe your backup with your backup (alternatively: is backing it up actually useful assuming git? there is also the `temp_changelog` so seems like there are actually three versions of it during the lifetime of the script where two could be sufficient?)

**Review by Cal-L — COMMENTED** [review]
Update description instructions from `yarn changeset-documentation` -> `yarn changeset-changelog`

File: .github/workflows/create-release-pr.yml [hunk 5]
```diff
@@ -50,4 +50,4 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
-          ./scripts/create-release-pr.sh ${{ github.event.inputs.semver-version }}
+          ./scripts/create-release-pr.sh ${{ github.event.inputs.semver-version }}
```
Comment (sethkfman) [inline .github/workflows/create-release-pr.yml:53]:
```suggestion
          ./scripts/create-release-pr.sh ${{ github.event.inputs.semver-version }}
          
```

File: scripts/changelog-csv.sh (hunk 3 above)
Comment (sethkfman) [inline scripts/changelog-csv.sh:75]:
```suggestion
echo 'CHANGELOG updated'

```

File: scripts/generate-rc-commits.js [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+ * This script is used to filter and group commits by teams based on unique commit messages.
+ * It takes two branches as input and generates a CSV file with the commit hash, commit message, author, team, and PR link.
+ * The teams and their members are defined in the 'authorTeams' object.
+ *
+ * Command to run the script: node development/generate-rc-commits.js origin/branchA origin/branchB
+ * Output: the generated commits will be in a file named 'commits.csv'.
+ */
+
+// JSON mapping authors to teams
+const authorTeams = {
```
Comment (legobeat) [inline scripts/generate-rc-commits.js:13]: Can this be fetched from GH API rather than hardcoded?

**Pull request closed without merging**